nav2_amcl/src/amcl_node.cpp

```diff
@@ -782,6 +830,21 @@ bool AmclNode::shouldUpdateFilter(const pf_vector_t pose, pf_vector_t & delta)
   bool update = fabs(delta.v[0]) > d_thresh_ ||
     fabs(delta.v[1]) > d_thresh_ ||
     fabs(delta.v[2]) > a_thresh_;
+
+
+  if (recovery_scan_count_ > 0) {
+    RCLCPP_INFO(get_logger(), "Performing Nomotion Update");
+    force_update_ = true;
+    recovery_scan_count_--;
+  }
+  else if (recovery_run_count_ > 0) {
+    recovery_scan_count_ = recovery_scan_updates_per_run_;
+    noise_only_update_ = true;
+    force_update_ = true;
+    recovery_scan_count_--;
```

Comment:
This will cause `recovery_scan_count_` to become negative as this will only be executed if `recovery_scan_count_` is <= 0, would this be a problem?

Reply:
You are perfectly right, but since the value is only used in the condition you mentioned it should be fine.
If you come up with a cleaner way to write this I am happy to change it, though.